lib/puppet/parser/functions/fingerprint_formatter.rb

```diff
@@ -0,0 +1,23 @@
+# A custom function to take fingerprints and format them for comparing against `apt-key fingerprint`
+# fingerprint_formatter("6F6B15509CF8E59E6E469F327F438280EF8D349F") == "6F6B 1550 9CF8 E59E 6E46  9F32 7F43 8280 EF8D 349F"
+module Puppet::Parser::Functions
+  newfunction(:fingerprint_formatter, type: :rvalue) do |args|
+    fingerprint = args[0]
+
+    if fingerprint.nil?
+      raise Puppet::ParseError, 'fingerprint_formatter requires a fingerprint to format as an argument.'
+    end
+
+    fingerprint_broken_up = []
+    beginning, ending = 0, 3
+    10.times do
+      fingerprint_broken_up << fingerprint.split('')[beginning..ending]
+      beginning += 4
+      ending += 4
+    end
```

Comment:
Maybe I'm missing something but:
```
print = '6F6B15509CF8E59E6E469F327F438280EF8D349F'
> "6F6B15509CF8E59E6E469F327F438280EF8D349F"
print.split(//).each_slice(4).to_a.map(&:join).join(' ')
=> "6F6B 1550 9CF8 E59E 6E46 9F32 7F43 8280 EF8D 349F"
```

Reply:
(there might be a prettier way to do that on a string without breaking it into a an array with single-char elements tho)

Reply:
❤️  thank you. Not missing anything, I just can't ruby.